src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -234,6 +246,19 @@ private static void LogErrors(ScriptHost host)
             }
         }
 
+        /// <summary>
+        /// Handler for any file changes under the root script path
+        /// </summary>
+        private void OnScriptFileChanged(object sender, FileSystemEventArgs e)
+        {
+            if (_restartDelayTokenSource != null && !_restartDelayTokenSource.IsCancellationRequested)
+            {
+                // if we're currently awaiting an error/restart delay
```

Comment:
Any such changes will break out of the delay, resuming the restart immediately. This is to allow the delay to backoff in steady state app scenarios, while retaining development time responsiveness.